### ISSUE TITLE
Problem: "/" route does not use infinite scroll

### ISSUE THREAD

**Issue by gazhayes** [Blockrazor/blockrazor#543]
Problem: even in the early stages, Blockrazor has to be able to deal with 1000+ currencies and not break. From a UI perspective, 1000 currencies on the home route is not going to perform very well. 

Also from a UI perspective, it's a bad idea to make users click "more" or something at the end of the page to keep loading more, this should be done automatically when hitting the bottom of the page.

**Comment by janat08** [comment]
doing this.

**Comment by gazhayes** [comment]
Can you make sure your other issues are completed before starting on new ones, unless the new one is related to an existing one you are working on.

**Comment by janat08** [comment]
doing this.

**Comment by gazhayes** [comment]
This also has the same search issues as #548.

**Comment by janat08** [comment]
I'll keep subscription on for fast render, and then stop rendering results from sub disabling it once the method is loaded? For this, I'd have to start passing data in from "smart component" to "reusable components".

**Comment by gazhayes** [comment]
Is fast render still going to try and send 3000 currencies in with the first payload if you do that?

**Comment by janat08** [comment]
I guess it will only send as many as we need for a page.

**Comment by gazhayes** [comment]
OK. Well, I guess you should experiment and see what happens

**Comment by janat08** [comment]
List of things to consider to add:
- show the item number being seen out of how many
- back to the top button
- all loaded styling
- if 2x limit is smaller then what could be seen in window it won't load anything else

**Comment by gazhayes** [comment]
> show the item number being seen out of how many

I was thinking about  putting some statistics in the header nav instead (e.g. xx projects currently listed), something like that. Maybe the UI guys will have a nice idea about how to give users the impression that there are a lot of currencies (when there actually are a lot).

> back to the top button

Yes I'd say this is pretty important

> all loaded styling

Can you explain what you mean?

> if 2x limit is smaller then what could be seen in window it won't load anything else

I don't quite understand, can you explain this in a different way?

**Comment by janat08** [comment]
> all loaded styling
when user reaches bottom, I just stuck a text explaining that they've reached the end of list.

> show the item number being seen out of how many
a lot like in meteor forums' topics you'll have an indicator that shows what item you're viewing and how many are left

> if 2x limit is smaller then what could be seen in window it won't load anything else
the limit is 15, if you have UHD screen it will load 30 items because it tries to fill the screen, if you're big enough (or you've scaled things down enough) it will load the 30 items and user won't be able to scroll thus the library won't detect that it's actually not filling the screen.

**Comment by gazhayes** [comment]
> the limit is 15, if you have UHD screen it will load 30 items because it tries to fill the screen, if you're big enough (or you've scaled things down enough) it will load the 30 items and user won't be able to scroll thus the library won't detect that it's actually not filling the screen.

OK, that has to be fixed. Do you think you're able to fix that? If not, can you log a separate issue?

**Comment by janat08** [comment]
If you want to be basic about it, you could just do rough math that looks at the current layout and calculates appropriate beginning limit and increments for user. The library used for scroll tracking is fairly elaborate, and it likely is able to deal with this. I already explored it and decided to skip the issue. If someone has experience with ScrollMagic they should have a go, otherwise, I'd be glad to explore further.

**Comment by gazhayes** [comment]
OK. I guess we should just actually test it on UHD screens and see what happens. I'll close this for now as complete, and open a new issue if there's a problem.
